src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -192,12 +188,15 @@ protected void LookupMembersInternal(LookupResult result, NamespaceOrTypeSymbol
                 type = extendedType;
             }
 
-            getCompatibleExtensions(this, type, compatibleExtensions, originalBinder, basesBeingResolved);
+            getCompatibleExtensions(this, type, compatibleExtensions, originalBinder, basesBeingResolved, skipType);
 
             // Sort extensions from more specific to less specific
             var tempUseSiteInfo = new CompoundUseSiteInfo<AssemblySymbol>(useSiteInfo);
             compatibleExtensions.Sort((x, y) => isMoreSpecificExtension(x, y) ? -1 : 1); // Note: captures tempUseSiteInfo
-            useSiteInfo.MergeAndClear(ref tempUseSiteInfo);
+            if (tempUseSiteInfo.AccumulatesDiagnostics)
+            {
+                useSiteInfo.MergeAndClear(ref tempUseSiteInfo);
```

Comment:
>MergeAndClear [](http://example.com/codeflow?start=28&length=13)

It might be better to adjust behavior of ```MergeAndClear```. However, I do not suggest doing this in context of this PR. Consider opening an issue to follow up during quality milestone